include/nbl/builtin/hlsl/type_traits.hlsl

```diff
@@ -744,28 +743,27 @@ struct float_of_size
 {
     using type = void;
 };
-
 template<>
 struct float_of_size<2>
 {
     using type = float16_t;
 };
-
 template<>
 struct float_of_size<4>
 {
     using type = float32_t;
 };
-
 template<>
 struct float_of_size<8>
 {
     using type = float64_t;
 };
-
 template<uint16_t bytesize>
 using float_of_size_t = typename float_of_size<bytesize>::type;
 
+template<typename T, int N>
+struct extent<vector<T, N>, 0> : integral_constant<uint64_t, N> {};
+
```

Comment:
vectorial?

btw matrix should report same as a `extent<T[N][M]>`

Reply:
we don't consider `Vectorial` in `type_traits.hlsl`, no?

Reply:
the partial spec for Vectorial could be defined just after where `Vectorial` is

however a partial spec with a constraint requires a constrainable primary template def AFAIK?